File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.cs
public class Solution {
    public int FindLucky(int[] arr) 
    {
        var fre = new Dictionary<int,int>();
        int ans = -1;
        foreach (int item in arr)
        {
            if (!fre.ContainsKey(item))
                fre[item] = 1;
            else
                fre[item]++;
        }
        foreach (var item in fre)
        {
            var key = item.Key;
            var value = item.Value;
            if (key == value)
                ans = Math.Max(ans, key);
        }
        return ans;
    }
}

File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) 
    {
        map <int,int> fre;
        for (auto item : arr)
            fre[item]++;
        
        int ans = -1;
        for (auto item : fre)
        {
            int key = item.first;
            int value = item.second;

            if (key == value)
                ans = key;
        }
        return ans;
    }
};